pulsar-common/src/main/java/org/apache/pulsar/common/util/keystoretls/KeyStoreSSLContext.java

```diff
@@ -159,8 +159,10 @@ public SSLContext createSSLContext() throws GeneralSecurityException, IOExceptio
                     : TrustManagerFactory.getInstance(tmfAlgorithm);
             KeyStore trustStore = KeyStore.getInstance(trustStoreTypeString);
             char[] passwordChars = trustStorePassword.toCharArray();
-            try (FileInputStream inputStream = new FileInputStream(trustStorePath)) {
-                trustStore.load(inputStream, passwordChars);
+            if (!Strings.isNullOrEmpty(trustStorePath)) {
```

Comment:
> What is the implication of an empty value resulting in skipping the load here? 

When we use the certificate from the public CA, we don't need to set up the trust store.

Reply:
Based on quickly reading through the code for conscrypt, I am wondering if this PR will work as described. Have you tested it? I see the following code in conscrypt:

```
    public void engineInit(KeyStore ks) throws KeyStoreException {
        if (ks != null) {
            keyStore = ks;
        } else {
            keyStore = Platform.getDefaultCertKeyStore();
        }
    }
```
That indicates to me that we should pass `null` to the `trustManagerFactory.init(trustStore);` instead of passing a created but not yet initialized `trustStore`.

Reply:
Good catch! You are right.

I checked the JDK codebase, see 
```
    @Override
    protected TrustManager[] engineGetTrustManagers() {
        if (!isInitialized) {
            throw new IllegalStateException(
                        "TrustManagerFactoryImpl is not initialized");
        }
        return new TrustManager[] { trustManager };
    }
```
We must init the `TrustManagerFactory`.

Reply:
This part has been improved.

Reply:
For reference, calling `trustManagerFactory.init(null)`, calls: 

```
    public final void init(KeyStore ks) throws KeyStoreException {
        factorySpi.engineInit(ks);
    }
```

And that calls the following in conscrypt:

```
    @Override
    public void engineInit(KeyStore ks) throws KeyStoreException {
        if (ks != null) {
            keyStore = ks;
        } else {
            keyStore = Platform.getDefaultCertKeyStore();
        }
    }
```

https://github.com/google/conscrypt/blob/ca6485b689219323e91d3bbe0410715b5b2d687f/common/src/main/java/org/conscrypt/TrustManagerFactoryImpl.java#L55-L65

That is why I suggested using `null` as the `KeyStore`. I haven't found documentation to suggest that this is the general implementation though.

Reply:
Your share is a good idea. But when the trust store file path is empty, we don't need to create the `TrustManagerFactory`.

> TrustManager[] - the sources of peer authentication trust decisions or null

Reply:
Great, that makes sense, and looks right to me.